### ISSUE TITLE
this.checklist is undefined Stack-Trace

### ISSUE THREAD

**Issue by 4ny0n3** [zourtney/angular-checklist#3]
Hi,

I followed all the steps of your tutorial, but I get this error message in my Browser.
`ERROR TypeError: this.checklist is undefined
Stack-Trace:
../../../../angular-checklist/dist/directives/checklist.directive.js/ChecklistDirective.prototype.ngOnChanges@http://localhost:4200/vendor.bundle.js:16:9
checkAndUpdateDirectiveInline@http://localhost:4200/vendor.bundle.js:56953:9
checkAndUpdateNodeInline@http://localhost:4200/vendor.bundle.js:58452:17
checkAndUpdateNode@http://localhost:4200/vendor.bundle.js:58391:16
debugCheckAndUpdateNode@http://localhost:4200/vendor.bundle.js:59252:38
debugCheckDirectivesFn@http://localhost:4200/vendor.bundle.js:59193:13
View_Question1Component_3/<@ng:///AppModule/Question1Component.ngfactory.js:775:5
debugUpdateDirectives@http://localhost:4200/vendor.bundle.js:59178:12
checkAndUpdateView@http://localhost:4200/vendor.bundle.js:58358:5
callViewAction@http://localhost:4200/vendor.bundle.js:58723:21
execEmbeddedViewsAction@http://localhost:4200/vendor.bundle.js:58681:17
checkAndUpdateView@http://localhost:4200/vendor.bundle.js:58359:5
callViewAction@http://localhost:4200/vendor.bundle.js:58723:21
execComponentViewsAction@http://localhost:4200/vendor.bundle.js:58655:13
checkAndUpdateView@http://localhost:4200/vendor.bundle.js:58364:5
callViewAction@http://localhost:4200/vendor.bundle.js:58723:21
execEmbeddedViewsAction@http://localhost:4200/vendor.bundle.js:58681:17
checkAndUpdateView@http://localhost:4200/vendor.bundle.js:58359:5
callViewAction@http://localhost:4200/vendor.bundle.js:58723:21
execComponentViewsAction@http://localhost:4200/vendor.bundle.js:58655:13
checkAndUpdateView@http://localhost:4200/vendor.bundle.js:58364:5
callWithDebugContext@http://localhost:4200/vendor.bundle.js:59578:39
debugCheckAndUpdateView@http://localhost:4200/vendor.bundle.js:59118:12
../../../core/@angular/core.es5.js/</ViewRef_.prototype.detectChanges@http://localhost:4200/vendor.bundle.js:56289:9
../../../core/@angular/core.es5.js/</ApplicationRef_.prototype.tick/<@http://localhost:4200/vendor.bundle.js:50927:58
../../../core/@angular/core.es5.js/</ApplicationRef_.prototype.tick@http://localhost:4200/vendor.bundle.js:50927:13
next/<@http://localhost:4200/vendor.bundle.js:50804:100
../../../../zone.js/dist/zone.js/</</ZoneDelegate.prototype.invoke@http://localhost:4200/polyfills.bundle.js:2806:17
onInvoke@http://localhost:4200/vendor.bundle.js:50010:24
../../../../zone.js/dist/zone.js/</</ZoneDelegate.prototype.invoke@http://localhost:4200/polyfills.bundle.js:2805:17
../../../../zone.js/dist/zone.js/</</Zone.prototype.run@http://localhost:4200/polyfills.bundle.js:2556:24
../../../core/@angular/core.es5.js/</NgZone.prototype.run@http://localhost:4200/vendor.bundle.js:49941:54
next@http://localhost:4200/vendor.bundle.js:50804:70
../../../core/@angular/core.es5.js/</EventEmitter.prototype.subscribe/schedulerFn<@http://localhost:4200/vendor.bundle.js:49755:36
../../../../rxjs/Subscriber.js/SafeSubscriber.prototype.__tryOrUnsub@http://localhost:4200/vendor.bundle.js:4688:13
../../../../rxjs/Subscriber.js/SafeSubscriber.prototype.next@http://localhost:4200/vendor.bundle.js:4635:17
../../../../rxjs/Subscriber.js/Subscriber.prototype._next@http://localhost:4200/vendor.bundle.js:4575:9
../../../../rxjs/Subscriber.js/Subscriber.prototype.next@http://localhost:4200/vendor.bundle.js:4539:13
../../../../rxjs/Subject.js/Subject.prototype.next@http://localhost:4200/vendor.bundle.js:4283:17
../../../core/@angular/core.es5.js/</EventEmitter.prototype.emit@http://localhost:4200/vendor.bundle.js:49741:54
checkStable@http://localhost:4200/vendor.bundle.js:49975:13
onHasTask@http://localhost:4200/vendor.bundle.js:50023:21
../../../../zone.js/dist/zone.js/</</ZoneDelegate.prototype.hasTask@http://localhost:4200/polyfills.bundle.js:2859:21
../../../../zone.js/dist/zone.js/</</ZoneDelegate.prototype._updateTaskCount@http://localhost:4200/polyfills.bundle.js:2879:17
../../../../zone.js/dist/zone.js/</</Zone.prototype._updateTaskCount@http://localhost:4200/polyfills.bundle.js:2703:17
../../../../zone.js/dist/zone.js/</</Zone.prototype.runTask@http://localhost:4200/polyfills.bundle.js:2623:25
drainMicroTaskQueue@http://localhost:4200/polyfills.bundle.js:3010:25
../../../../zone.js/dist/zone.js/</</ZoneTask.invokeTask@http://localhost:4200/polyfills.bundle.js:2917:21
invokeTask@http://localhost:4200/polyfills.bundle.js:3785:9
globalZoneAwareCallback@http://localhost:4200/polyfills.bundle.js:3803:17`

Thank you in advance.

**Comment by zourtney** [comment]
Sorry I didn't get back to you yesterday. Were you able to figure it out? If not, I'd be happy to help. 

Just post the relevant source code, so I have a little more to go off of. An online editor, like JSFiddle, would be even better.

**Comment by 4ny0n3** [comment]
Unfortunately I was ill the past days.
This is the relevant source code.
[src.zip](https://github.com/zourtney/angular-checklist/files/1227261/src.zip)
There is a new message prompted by Visual Studio code:

> file: 'file:///d%3A/Angular/testing/node_modules/angular-checklist/tsconfig.json'
severity: 'Warnung'
message: 'Problems loading reference 'http://json.schemastore.org/tsconfig': Unable to load schema from 'http://json.schemastore.org/tsconfig': Unable to to connect to http://json.schemastore.org/tsconfig. Error: connect ETIMEDOUT 137.117.17.70:80'
at: '1,1'
source: ''

""

**Comment by 4ny0n3** [comment]
Any news about my request?

**Comment by zourtney** [comment]
Sorry for the delay! I'll look into it tomorrow morning.

**Comment by zourtney** [comment]
Well, I've been unable to reproduce the error stated above. Definitely to my inexperience with SystemJS. (I didn't see the actual checklist code in `src.zip`, or an inclusion line in `systemjs.config.js`. If you have a more complete example, that would help me.)

Still, I think the original error comes from the fact that the `@Input('checklist')` value is undefined when the component initially mounts. Set it to an empty array instead, and let me know if it works. For example:

```ts
@Component({
  selector: 'demo',
  template: `
    <input type="checkbox" [checklist]="myChecklistArray" checklistValue="listVal1" />
  `,
})
export class AppComponent {
  myChecklistArray = [];
}
```

If that's the only issue, we could make the `ngOnChanges` check null-safe by changing it to something like:

```ts
ngOnChanges() {
  this.isChecked = this.checklist && this.checklist.indexOf(this.checklistValue) >= 0;
}
```

**Comment by 4ny0n3** [comment]
If have found the Error, I had to change `interests : String[];` to `interests = [];`

Thank you for your time and effort

**Comment by zourtney** [comment]
You bet! Sorry I didn't catch that the first time.  Let me know if you have any further issues.

**Comment by 4ny0n3** [comment]
Unfortunately I have to bother you again.
When I compile the program for production use with "ng build --prod" and upload it to the server a similar message is prompted.

> ERROR TypeError: this.checklist is undefined
Stack-Trace:
aZoX/o</t.prototype.ngOnChanges@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:136494
Cn@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:27139
pr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41979
cr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41512
Dr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:47643
u/<@http://192.168.38.201/main.3ebea605a01c8f789f99.bundle.js:1:6334
updateDirectives@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:45576
ur@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41195
wr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:44035
br@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:43433
ur@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41330
/oeL/</Gs</t.prototype.detectChanges@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:116561
/oeL/</Sa</e.prototype.tick/<@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:86709
/oeL/</Sa</e.prototype.tick@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:86670
/oeL/</Sa</e.prototype._loadComponent@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:86271
/oeL/</Sa</e.prototype.bootstrap@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:86054
/oeL/</Pa</e.prototype._moduleDoBootstrap/<@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:83454
/oeL/</Pa</e.prototype._moduleDoBootstrap@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:83404
/oeL/</Pa</e.prototype._bootstrapModuleFactoryWithZone/</</<@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:82925
eFQL/</</</l</e.prototype.invoke@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:40510
onInvoke@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:9420
eFQL/</</</l</e.prototype.invoke@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:40437
eFQL/</</</u</r.prototype.run@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:35662
l/<@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:45434
eFQL/</</</l</e.prototype.invokeTask@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:41185
onInvokeTask@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:9332
eFQL/</</</l</e.prototype.invokeTask@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:41098
eFQL/</</</u</r.prototype.runTask@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:36353
o@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:33425
  vendor.bb6d96105ff946c8dcee.bundle.js:1:4648
ERROR TypeError: this.checklist is undefined
Stack-Trace:
aZoX/o</t.prototype.ngOnChanges@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:136494
Cn@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:27139
pr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41979
cr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41512
Dr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:47643
u/<@http://192.168.38.201/main.3ebea605a01c8f789f99.bundle.js:1:6361
updateDirectives@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:45576
ur@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41195
wr@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:44035
br@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:43433
ur@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:41330
/oeL/</Gs</t.prototype.detectChanges@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:116561
/oeL/</Sa</e.prototype.tick/<@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:86709
/oeL/</Sa</e.prototype.tick@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:86670
next/<@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:84573
eFQL/</</</l</e.prototype.invoke@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:40510
onInvoke@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:9420
eFQL/</</</l</e.prototype.invoke@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:40437
eFQL/</</</u</r.prototype.run@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:35662
/oeL/</ha</t.prototype.run@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:78804
next@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:84550
/oeL/</da</e.prototype.subscribe/o<@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:77318
T14+/c</e.prototype.__tryOrUnsub@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:133400
T14+/c</e.prototype.next@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:132675
T14+/u</e.prototype._next@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:131717
T14+/u</e.prototype.next@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:131400
rlar/p</e.prototype.next@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:294895
/oeL/</da</e.prototype.emit@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:77077
ot@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:9028
onHasTask@http://192.168.38.201/vendor.bb6d96105ff946c8dcee.bundle.js:1:9565
eFQL/</</</l</e.prototype.hasTask@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:41504
eFQL/</</</l</e.prototype._updateTaskCount@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:41863
eFQL/</</</u</r.prototype._updateTaskCount@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:37945
eFQL/</</</u</r.prototype.runTask@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:36565
o@http://192.168.38.201/polyfills.0963b44a58b5526cfab7.bundle.js:1:33425<

**Comment by zourtney** [comment]
Hmm, I don't see any good reason why it would behave differently in a production build (unless it was using the previous code).

Anyway, I went ahead and published a new version, `1.2.3`, which makes the `isChecked` test null-safe. Let me know if this works any better for you.

**Comment by 4ny0n3** [comment]
Now it works!!

Thank your very much.
